File: ruby/fibtail.rb
def fibonacci(n, a, b)
  return a if n == 0
  return b if n == 1
  return fibonacci(n - 1, b, a + b)
end

begin
  sequences = ARGV[0].to_i
  puts "Ruby Fib: #{sequences}x"
  start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  output = fibonacci(sequences, 0, 1)
  finish = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  elapsed = (finish - start) * 1000
  puts "Result: #{output} | Time: #{elapsed.to_i}ms"
rescue Exception => e
  puts e
end

File: elixir/fibtail.exs
defmodule Fib do
  def fibonacci(n, a, b) do
    cond do
      n == 0 ->
        a
      n == 1 ->
        b
      true ->
        fibonacci(n - 1, b, a + b)
    end
  end
end

try do
  [arg1] = System.argv
  sequences = String.to_integer(arg1)
  IO.puts "Elixir Fib: #{sequences}x"
  start = :os.system_time(:millisecond)
  output = Fib.fibonacci(sequences, 0, 1)
  finish = :os.system_time(:millisecond)
  elapsed = (finish - start)
  IO.puts "Result: #{output} | Time: #{elapsed}ms"
rescue
  e -> IO.puts e
end
